### DISCUSSION TITLE
Extend a distinct zod instance

### DISCUSSION THREAD

**Discussion by qw-in** [asteasolutions/zod-to-openapi#175] [Q&A] [answered]
Thanks for this, cool library!

We use `zod` for a lot throughout the codebase, and I'd prefer to keep the openapi bits contained. Wondering how feasible something like the following would be:

```ts
// somewhere.ts

import { z } from "zod":
import { createExtendedZod } from ".";

export const oz = createExtendedZod(z);

// elsewhere.ts

import { oz } from "./somewhere"

const schema = oz.string().openapi(/* ... */);
```

Thanks!

**Comment by AGalabov** [comment, marked as answer]
@qw-in thank you for the kind word. If I understood correctly you just want to do the extending in a single file and reexport it. You do not need `createExtendedZod` for that. What we've done for a project where we use it is:

```ts
// utils/zod.ts
import { z } from 'zod';

extendZodWithOpenApi(z);

export { z }; // If you want to "rename it" you can just use `export { z as oz };`
```

And then we import it as:

```ts
import { z } from '<path-here>/utils/zod'
```

We've also added an eslint rule not to import from `'zod'` directly. 

Is this what you intended and does this solve it for you?

**Reply by qw-in** [reply]
@AGalabov ahh, thanks for the clarification!

I saw the types (`.d.ts`) and assumed it was globally patching zod. Perfect, I'll give it a try next week. Cheers
